specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

```diff
@@ -828,6 +828,13 @@
         "storeSettings": {
           "$ref": "#/definitions/StoreReadSettings",
           "description": "Avro store settings."
+        },
+        "additionalColumns": {
```

Comment:
Adding new properties to the response of an API is classified as a breaking change. Please make these changes in a new api-version.

https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/guestaccess.aspx?e=xEcppo&share=EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw

Reply:
We had a discussion with ARM team in August that this kind of change in ADF swagger is not considered as a breaking change. Below is quote from the email between us:

_Regarding to the “definition” of breaking change, I think ADF has a looser definition than that of ARM, and looks like it’s allowable from definition of the spec :  RestAPI_BreakingChangeGuide.docx_

_“Teams MAY define backwards compatibility as their business needs require. For example, Azure defines the addition of a new JSON field in a response to be not backwards compatible. Office 365 has a looser definition of backwards compatibility and allows JSON fields to be added to responses.”_

_Back to ADF business needs, it is an data integration service which support 80+ data stores and certain number of other compute system as well, we model these data stores as **LinkedService** (with a “type” property to identify the actual data store type), and the each piece of data in these data stores as **DataSet** (with a “type” property to identify the actual data type), while we have another model object called **Pipeline** which refer to **LinkedServices** and **DataSets** to define activities to process these data. As you can see, in this model design, whenever we add a new type of **LinkedService** and **DataSet**, it will result in adding new properties of the API, but these new properties won’t impact the main concept of our model. Moreover, if we always upgrade API version in this case, it will be very difficult for customer to figure out which API version support what data stores given we have large number of  data stores and we add this data store support very quickly. So per these business requirements,  we decided not to add major version API for new connectors._

_Regarding to the swagger review process, I would expect both ARM and Swagger review team can understand this, and we don’t need to explain these details for each single swagger review._

Reply:
Can you forward me the email thread regarding this? rbenson@microsoft.com

The idea that RP teams can define backwards compatibility it supposed to enable teams to be more restrictive than ARM on changes, not less. 

You'll notice the O365 reference is omitted in the ARM breaking changes doc and was replaced by Azures high level view of a breaking change:
"Anything that would violate the Principle of Least Astonishment is considered a breaking change in Azure. Below are some concrete examples of what constitutes a breaking change. In the below breaking change scenarios, the API version **must** be changed."

The O365 reference comes from the following doc which describes the all up Microsoft breaking change definition:
https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change

When reading the Microsoft docs, Azure is the business and ADF is a component of Azure.

This isn't just a random requirement we added. If you introduce new LinkedService kinds in the same api version, the existing SDK could start failing. ie If one of your new LinkedServices is returned, it will fail to deserialize. If you introduce it in a new API version, the old SDK wouldn't see the new LinkedServices and would continue to work.

If you have an exception for this, you will have to explain this to every reviewer as this is technically a breaking change according to Azure.